commons/zenoh-protocol/src/core/cowstr.rs

```diff
@@ -0,0 +1,105 @@
+use core::fmt::{Debug, Display, Formatter};
```

Comment:
Should the `cowstr` part of `zenoh-collections` instead of `zenoh-protocol`?

Reply:
Meh, I also hesitated, not quite a collection either, we can move it if you want

Reply:
You are right, I'm also hesitating on that. But I see it more generic than just the protocol...